src/mlpack/core/data/random_regression_generator.hpp

```diff
@@ -0,0 +1,210 @@
+/**
+ * @file random_regression_generator.hpp
+ * @author Ali Hossam
+ *
+ * Implementation of a regression data generator with random features and error 
+ * distribution.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license. You should have received a copy of the
+ * 3-clause BSD license along with mlpack. If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_CORE_DATA_RANDOM_REGRESSION_GENERATOR_HPP
+#define MLPACK_CORE_DATA_RANDOM_REGRESSION_GENERATOR_HPP
+
+#include <mlpack.hpp>
+
+namespace mlpack {
+namespace data {
+
+
+/**
+ * Error distribution types.
+ */
+enum class ErrorType
+{
+  NormalDist, /**< Normal distribution.  */
+  GammaDist   /**< Gamma distribution. */
+};
+
+struct NormalDistParams
+{
+  double mu;  /**< Mean. */
+  double std; /**< Standard deviation. */
+
+  NormalDistParams(double mu, double std) : mu(mu), std(std) {}
+};
+
+struct GammaDistParams
+{
+  double alpha;
+  double beta; 
+
+  GammaDistParams(double alpha, double beta) : alpha(alpha), beta(beta) {}
+};
+
+struct ErrorParams
+{
+  ErrorType type; /**< Type of error distribution. */
+
+  union
+  {
+    NormalDistParams normalParams; /**< Parameters for normal distribution. */
+    GammaDistParams gammaParams;   /**< Parameters for gamma distribution. */
+  };
+  
+  ErrorParams(ErrorType type) : type(type) {};
+};
```

Comment:
All of these are not required, since armadillo handles all of these. Why then create empty structs that do not do anything?

Reply:
Hi @shrit 
Thank your for your feedback,
"I thought about adding more error distribution options in the future, like Laplace and Discrete distributions. However, I couldn't find a straightforward way for users to input only the needed parameters for their chosen distribution (like mu and std for normal distribution, alpha and beta for gamma distribution), apart from the current method.

I also considered letting users pass an array with parameters for each distribution, using an enumeration for clarity:

```cpp
enum ErrorType {
    Normal, 
    Gamma,
    // ...
};

int main() {
    // Normal distribution - mu -> params[0], std -> params[1]
    double params[] = {2.0, 3.0};
    RegressionDataGenerator generator(100, 1, ErrorType::Normal, params);

    // Gamma distribution - alpha -> params[0], beta -> params[1]
    double params[] = {2.0, 3.0};
    RegressionDataGenerator generator(100, 1, ErrorType::Gamma, params);

    // Additional distributions
    // ...
}
```

But I felt that the first method is more user-friendly, what do you think?

Reply:
You can just use template parameters and overload the constructors, look at other functions, for example tree classes can be a good example to inspire what are you trying to achieve

Reply:
Kindly recheck when possilbe